src/service/url-replacements-impl.js

```diff
@@ -163,6 +164,14 @@ export class UrlReplacements {
       return removeFragment(info.sourceUrl);
     }));
 
+    this.setAsync_('SOURCE_URL', () => {
+      return getTrackImpressionPromise().then(
+        this.getDocInfoValue_.bind(this, info => {
```

Comment:
add a getter on `source_url`. I also removed `url` since it's not used anywhere.